### DISCUSSION TITLE
/condense for melon slices

### DISCUSSION THREAD

**Discussion by radiant-ai** [Zrips/CMI#5304] [Feature Requests]
A couple of players requested an ability to turn melon slices into melon blocks using `/condense` command. This is it, nothing more. If it is not worthy to add to CMI don't do it, I will do it myself, doesn't sound difficult, right? :)

**Comment by Zrips** [comment]
Simply create custom recipe which converts slices into melons and you will have option like that

**Reply by radiant-ai** [reply]
This recipe already exists. I am asking about /condense.

**Reply by MaHl111** [reply]
This seems pretty stupid to me, why would you condense slices in to melons. Melon slices are not even good source of food, so it makes even less sense to have that many slices you need to condense them inside of your inventory.

**Edit:** Only proves how useless melons really are and fact I always avoided them because they aren't useful in practice for anything else except trading (which also have better alternatives) and Glistering Melon, used for potion making.

**Reply by radiant-ai** [reply]
> This seems pretty stupid to me, you can't make slices in to melon by default, why would you condense slices in to melons. Melon slices are not even good source of food, so it makes even less sense to have that many slices you need to condense them inside of your inventory.

https://minecraft.gamepedia.com/Melon
![image](https://user-images.githubusercontent.com/43506080/111893065-29a6d600-89ce-11eb-85ac-ecd0e270e2d2.png)

**Reply by radiant-ai** [reply]
And I do not request, I just suggest, I already coded this even before asking, here is the code, it literary took 3 minutes to write and 2 to test. No? Just close the discussion.
```
if (sender instanceof Player) {
            Player p = (Player) sender;
            PlayerInventory inv = p.getInventory();
            int slices = 0;
            for (int i = 0; i < inv.getSize(); i++) {
                ItemStack is = inv.getItem(i);
                if (is != null && is.getType().equals(Material.MELON_SLICE)) {
                    slices += is.getAmount();
                    inv.setItem(i, null);
                }
            }
            int melons = slices / 9;
            slices = slices % 9;
            inv.addItem(new ItemStack(Material.MELON, melons));
            Map map = inv.addItem(new ItemStack(Material.MELON_SLICE, slices));
            if (!map.isEmpty()) {
                p.getLocation().getWorld().dropItemNaturally(p.getLocation(), new ItemStack(Material.MELON_SLICE, slices));
            }
            p.sendMessage(Lang.getLang("misc_melon"));
        }
        return false;
        ```

**Reply by 456dev** [reply]
i may be wrong, but i though that condense will auto detect recipes that have a forwards and reverse version.
so you may need melon block -> melon slice, then it works?
again, this is off memory, which might be wrong
